### PR TITLE
fewer github builds

### PR DESCRIPTION
**Pull request by DanGrayson**

### CODE REVIEW

**Comment by mahrud** [comment]
I think you should put cmake-macos-latest-clang10 back in. Both clang builds were nontrivial to get working, and I would like them not to break unintentionally.

If you'd like to reduce the number of builds, you should make a separate workflow that maybe runs 1 or 2 jobs, and perhaps when it is merged run a full workflow with 8 jobs.

File: .github/workflows/test_build.yml [hunk 1]
```diff
@@ -194,7 +182,7 @@ jobs:
         if: always()
         uses: actions/upload-artifact@v2
         with:
-           name: ${{ matrix.os }}-${{ matrix.build-system }}-${{ matrix.compiler }}-logs
+           name: ${{ matrix.build-system }}-${{ matrix.os }}-${{ matrix.compiler }}-logs
```
Comment (mahrud) [inline .github/workflows/test_build.yml:185]: No, this is very much incorrect and should be fixed!

This implies that caches for cmake-macos are compatible with caches for cmake-linux, which is not the case!

Comment (mahrud) [reply .github/workflows/test_build.yml:185]: Also, would be good if you allowed some time for review before merging.

Comment (mahrud) [reply .github/workflows/test_build.yml:185]: In particular, this is completely incompatible with this:
https://github.com/Macaulay2/M2/blob/1e47260755100936bafd60d64013672dedd3941f/.github/workflows/test_build.yml#L125-L129

Comment (mahrud) [reply .github/workflows/test_build.yml:185]: Ah, never mind, I forgot that the build cache key is defined separately from the log key. It's not a big deal, but it is confusing that the keys are ordered differently.

Comment (DanGrayson) [reply .github/workflows/test_build.yml:185]: I'd be happy to reorganize the lines concerned with caching, if you propose a change.

**Pull request merged**